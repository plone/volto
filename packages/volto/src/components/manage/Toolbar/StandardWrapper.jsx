import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { Icon } from '@plone/volto/components';
import backSVG from '@plone/volto/icons/back.svg';

const StandardWrapper = (props) => {
  function pull() {
    props.unloadComponent();
  }

  return (
    <div
      className={cx(`${props.componentName} pastanaga-menu`, {
        'has-inner-actions': props.hasActions,
      })}
      style={{
        flex: props.theToolbar.current
          ? `0 0 ${props.theToolbar.current.getBoundingClientRect().width}px`
          : null,
      }}
    >
      <header className="header pulled">
        <button onClick={pull}>
          <Icon name={backSVG} size="30px" />
        </button>
        <div className="vertical divider" />
        <h2>{props.componentTitle || props.componentName}</h2>
      </header>
      {props.children}
    </div>
  );
};

StandardWrapper.propTypes = {
  componentName: PropTypes.string.isRequired,
  componentTitle: PropTypes.string,
  unloadComponent: PropTypes.func.isRequired,
  loadComponent: PropTypes.func.isRequired,
  closeMenu: PropTypes.func.isRequired,
  hasActions: PropTypes.bool,
};

export default StandardWrapper;
