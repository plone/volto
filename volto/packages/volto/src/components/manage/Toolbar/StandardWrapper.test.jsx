import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import { MemoryRouter } from 'react-router-dom';

import StandardWrapper from './StandardWrapper';

const mockStore = configureStore();

describe('Toolbar Personal Tools component', () => {
  it('renders an Toolbar Personal Tools component', () => {
    const store = mockStore({
      content: {
        data: {
          '@type': 'Folder',
          is_folderish: true,
        },
      },
      intl: {
        locale: 'en',
        messages: {},
      },
    });
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <StandardWrapper
            componentName="profile"
            loadComponent={() => {}}
            unloadComponent={() => {}}
            closeMenu={() => {}}
            hasActions
            theToolbar={{
              current: { getBoundingClientRect: () => ({ width: '320' }) },
            }}
          />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
