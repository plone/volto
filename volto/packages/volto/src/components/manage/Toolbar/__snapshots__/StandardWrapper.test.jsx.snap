// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar Personal Tools component renders an Toolbar Personal Tools component 1`] = `
<div
  className="profile pastanaga-menu has-inner-actions"
  style={
    Object {
      "flex": "0 0 320px",
    }
  }
>
  <header
    className="header pulled"
  >
    <button
      onClick={[Function]}
    >
      <svg
        className="icon"
        dangerouslySetInnerHTML={
          Object {
            "__html": undefined,
          }
        }
        onClick={null}
        style={
          Object {
            "fill": "currentColor",
            "height": "30px",
            "width": "auto",
          }
        }
        viewBox=""
        xmlns=""
      />
    </button>
    <div
      className="vertical divider"
    />
    <h2>
      profile
    </h2>
  </header>
</div>
`;
